app/flags.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func reconcileEnvBool(key string, defaultVal bool) bool {
+	if _, set := os.LookupEnv(key); set {
```

Comment:
I worry that this logic is subtly confusing. I would expect that a command-line flag would override an environment variable, not the other way around.

I envision the "WTF IS GOING ON?!" moment as I run the code and my command-line flag changes don't take effect at all because something is hidden in the environment.